lisp/ess-custom.el

```diff
@@ -674,10 +674,12 @@ blink the filling region."
   :group 'ess-edit
   :type 'boolean)
 
-(defcustom ess-mode-silently-save t
-  "Non-nil means automatically save ESS source buffers before loading."
+(define-obsolete-variable-alias 'ess-mode-silently-save 'ess-silently-save "ESS 19.04")
+(defcustom ess-silently-save nil
```

Comment:
`ess-save-silently` would sound better and would be better namespaced.

Reply:
Why the default has been changed to nil? I would prefer it to be t by default. It's very unlikely that peopled wouldn't want it.

Reply:
@lionel- Yes, thanks, done.

@vspinu I changed it to nil since I don't think we should be saving files to disk without asking. It's also in line with what Emacs does (e.g. `compilation-ask-about-save`).

Reply:
I understand the logic and I am myself not entirely comfortable with it but, as always, it's a trade-off. If most or all our users will chose to auto-save then we shouldn't bother them with yet another awkward default. 

We can settle in the middle and check if auto-save mode is on and don't save if so.

Reply:
How about we just use `compilation-ask-about-save` instead of having our own? The purpose is so similar I can't imagine people wanting different behaviors.

Reply:
I like that idea.

Reply:
It's a non-generic setting and for a user wondering how to disable our "ask-for-save" there would be no way to figure out what to change except of going and reading the code. Our setting could default to compilation-ask-about-save in principle but I think auto-save-mode is more relevant still.  

We could have it set to 'auto and check both compilation-ask-about-save and auto-save-mode.

Reply:
Can you explain a bit why you think auto-save-mode is relevant here? If I understand correctly, auto-save-mode saves to separate files, it doesn't overwrite files like we're doing here. 

Are you thinking of `auto-save-visited-mode`?

Reply:
> Are you thinking of `auto-save-visited-mode`?

Yes, precisely.

Reply:
Gotcha. I didn't know about that newish mode until just now :-)

So now `ess-save-silently` defaults to `'auto`, which just means respect both `compilation-ask-about-save` and `auto-save-visited-mode`. I think this makes everyone happy, so I'll push to master later today or tomorrow unless I hear back.

Reply:
Thank you both for the above discussion .. and a very good resolution (from my reading)!